### PR TITLE
[YQ-1997] Refactor create external data source / external table schemeshard operations

### PR DESCRIPTION
**Pull request by Alnen**

### CODE REVIEW

File: ydb/core/tx/schemeshard/schemeshard__operation_create_external_data_source.cpp [hunk 1]
```diff
@@ -118,13 +35,13 @@ class TPropose: public TSubOperationState {
...
-        TTxState* txState = context.SS->FindTx(OperationId);
+        const TTxState* txState = context.SS->FindTx(OperationId);
         Y_ABORT_UNLESS(txState);
         Y_ABORT_UNLESS(txState->TxType == TTxState::TxCreateExternalDataSource);
 
-        auto pathId = txState->TargetPathId;
-        auto path = TPath::Init(pathId, context.SS);
-        TPathElement::TPtr pathPtr = context.SS->PathsById.at(pathId);
+        const auto pathId  = txState->TargetPathId;
+        const auto path    = TPath::Init(pathId, context.SS);
```
Comment (UgnineSirdis) [inline ydb/core/tx/schemeshard/schemeshard__operation_create_external_data_source.cpp:43]: Why? :)

File: ydb/core/tx/schemeshard/schemeshard__operation_create_external_data_source.cpp [hunk 2]
```diff
@@ -118,13 +35,13 @@ class TPropose: public TSubOperationState {
...
+        const auto pathPtr = context.SS->PathsById.at(pathId);
```
Comment (UgnineSirdis) [inline ydb/core/tx/schemeshard/schemeshard__operation_create_external_data_source.cpp:44]: Why auto is better than explicit `TPathElement::TPtr`?

File: ydb/core/tx/schemeshard/schemeshard__operation_create_external_data_source.cpp (hunk 1 above)
Comment (Alnen) [reply ydb/core/tx/schemeshard/schemeshard__operation_create_external_data_source.cpp]: Fixed formatting

File: ydb/core/tx/schemeshard/schemeshard__operation_create_external_data_source.cpp (hunk 2 above)
Comment (Alnen) [reply ydb/core/tx/schemeshard/schemeshard__operation_create_external_data_source.cpp]: Reverted change

**Review by UgnineSirdis — APPROVED** [review]

**Pull request merged**